File: aoc24-solutions/day01/day01p01.kt
import java.util.Scanner
import java.util.PriorityQueue
import kotlin.math.*

fun getDistance(h1: PriorityQueue<Int>, h2: PriorityQueue<Int>): Int {
    var total = 0
    for (i in 0 until h1.size) {
        val a = h1.poll()
        val b = h2.poll()
        total += abs(a - b)
    }
    return total
}

fun main() {
    val reader = Scanner(System.`in`)
    val h1 = PriorityQueue<Int>()
    val h2 = PriorityQueue<Int>()

    while (reader.hasNextLine()) {
        val line = reader.nextLine()
        val (first, last) = line.split("\\s+".toRegex())
        h1.add(first.toInt())
        h2.add(last.toInt())
    }
    println(getDistance(h1, h2))
}

File: aoc24-solutions/day01/day01p01.py
import sys
import heapq as heap


def get_distance(h1, h2) -> int:
    total = 0
    for _ in range(len(h1)):
        a = heap.heappop(h1)
        b = heap.heappop(h2)
        total += abs(a - b)
    return total


def main():
    h1, h2 = [], []
    for line in sys.stdin:
        a, b = line.split()
        heap.heappush(h1, int(a))
        heap.heappush(h2, int(b))
    return get_distance(h1, h2)

if __name__ == '__main__':
    print(main())
